### DISCUSSION TITLE
generating palabos input file using to_palabos

### DISCUSSION THREAD

**Discussion by nipinl** [PMEAL/porespy#776] [Q&A] [answered]
I was trying to generate palabos input file (.dat file with 0,1 & 2) from stl file. It is a sphere in cube 3d model as I intend to study flow through pack of sphere in palabos. I could generate series of bmp images from stl file using vedo (by binarize() and followed by slicing). Is there a similar option available in porespy?
Then, referring **loading_images.ipynb** I read the series of bmp images(50 Nos) using imread_collection to generate an im3d image of shape [50,50,3,50] with values ranging from 0.0 to 255.0. However, the dat file generated using the following command, _ps.io.to_palabos(im, tmp, solid=0)_ consists only of 0s. I found a link to an example of _to_palabos_  but it is broken. Any help in this is highly appreciated.
[files.zip](https://github.com/PMEAL/porespy/files/10362246/files.zip)

**Comment by nipinl** [comment]
I could do that by converting the series of bmp file to a binary tif file using fiji first, and then read into porespy using imageio.volread('binary.tif') followed by generating dat file by 
tmp = ('./palabos.dat') 
ps.io.to_palabos(imb, tmp, solid=0)
I suppose the conversion to binary can be done in porespy, but don't know exactly how.

**Comment by jgostick** [comment, marked as answer]
If you just want spheres, what about using either `rsa`, `pseudo_gravity_packing` or `pseudo_electrostatic_packing`?  Then you'll have 'voxel' images directly?  

As for our `to_palabos` function, I must admit that none of use have used it in several years, so although it passes our unit tests, the Palabos package may want something different by now. 

Lastly, conversion to binary is easy, you're over thinking it.  Just do: `im = bmp > 0`, and you'll have an image of `True` and `False`.

**Reply by nipinl** [reply]
Hi prof Jeff,
Thank you for suggesting me the options. I'll try those out. However, I plan to use non standard geometries too in the future. Thanks a lot for the tip for binarizing, now I need not go to a gui. palabos still work with the same input format and the porespy function **to_palabos** works flawlessly :-) :-)
